tests/jballerina-unit-test/src/test/resources/test-src/types/readonly/test_selectively_immutable_type.bal

```diff
@@ -1044,6 +1044,29 @@ function testFunctionWithReturnTypeAnyToReadonly() {
 
 function foo() returns readonly => 1;
 
+function testReadOnlyIntersectionWithNever() {
+    error? a = trap panicFn();
+
+    assertTrue(a is error);
+    assertEquality("err!", (<error> a).message());
+
+    record { never i?; int j; } & readonly b = {j: 1, "k": "str"};
```

Comment:
Shall we have some tests for theses cases?
-  closed records
-  open records with never rest field 
-  having field with `never? i`